protostar/commands/test/cheatcodes/given_cheatcode.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any
+
+from typing_extensions import Protocol
+
+from protostar.commands.test.fuzzing.strategy_descriptor import StrategyDescriptor
+from protostar.commands.test.fuzzing.strategy_selector import StrategySelector
+from protostar.commands.test.test_environment_exceptions import CheatcodeException
+from protostar.starknet.cheatcode import Cheatcode
+
+
+class StrategyLearnedException(BaseException):
+    """
+    An exception raised from the ``given`` cheatcode, indicating that the set of fuzzing strategies
+    has changed (fuzzer _learned_ a new strategy).
+
+    The expected behaviour is to let fuzzer catch this exception and restart fuzzing with new
+    set of input parameter strategies.
+    """
+
+
+class GivenCallable(Protocol):
+    def __call__(self, **kwargs: StrategyDescriptor):
+        ...
+
+
+class GivenCheatcode(Cheatcode):
+    def __init__(
+        self,
+        syscall_dependencies: Cheatcode.SyscallDependencies,
+        strategy_selector: StrategySelector,
+    ):
+        super().__init__(syscall_dependencies)
+        self.strategy_selector = strategy_selector
+
+    @property
+    def name(self) -> str:
+        return "given"
+
+    def build(self) -> GivenCallable:
+        return self.given
+
+    def given(self, **kwargs: Any) -> None:
+        # Early check that all parameters names are valid explicitly.
+        for param in kwargs:
+            self.strategy_selector.check_exists(param)
```

Comment:
If it's descriptive, what's the point of the comment?

Reply:
Hmm what I wanted to say is that this is an early check. Strategy selector would yell anyway, but I found the priority of invalid parameter name error higher than invalid strategy descriptor which is raised few lines below

Reply:
I don't understand see what is the benefit of this change too.

I would rename it though to
```
for param in kwargs:
            self.strategy_selector.assert_exists(param)
```

Reply:
@MaksymilianDemitraszek I am actually suggesting always to rename the other way -- from assert to check. The reason is that assert suggest throwing AssertionError which is not the case here. And also assertions suggest that they can be turned off and test invariants, while here we test user input.

Reply:
It's a fair point in python codebase. My logic was that assert suggests it throws error not returns a boolean.

Reply:
The benefit of the proposed change is lack of comment.

> Every time you write a comment, you should grimace and feel the failure of your ability of
expression.
Why am I so down on comments? Because they lie. Not always, and not intentionally,
but too often. The older a comment is, and the farther away it is from the code it describes,
the more likely it is to be just plain wrong. The reason is simple. Programmers can’t realistically maintain them.

Reply:
But we can remove the comment without introducing your proposed change. I dont see what the new name has to do with comments content.